addons/point_of_sale/static/src/backend/pos_payment_provider_cards/pos_payment_provider_cards.js

```diff
@@ -66,31 +70,36 @@ export class PosPaymentProviderCards extends Component {
         const provider = this.state.providers.find((p) => p.id === moduleId);
         if (provider) {
             this.props.record.update({
-                payment_method_type: "terminal",
-                use_payment_terminal: provider.selection,
+                use_payment_terminal: provider.use_payment_terminal,
+                use_payment_qr: provider.use_payment_qr,
                 name: provider.provider,
             });
         }
     }
 }
 
 // Selection, module_name, friendly name
-const providers = [
-    ["axepta_bnpp", "pos_iot_worldline", "Axepta BNP Paribas"],
-    ["six_iot", "pos_iot_six", "SIX"],
-    ["adyen", "pos_adyen", "Adyen"],
-    ["mercado_pago", "pos_mercado_pago", "Mercado Pago"],
-    ["razorpay", "pos_razorpay", "Razorpay"],
-    ["stripe", "pos_stripe", "Stripe"],
-    ["viva_com", "pos_viva_com", "Viva.com"],
-    ["worldline", "pos_iot_worldline", "Worldline"],
-    ["tyro", "pos_tyro", "Tyro"],
-    ["pine_labs", "pos_pine_labs", "Pine Labs"],
-    ["qfpay", "pos_qfpay", "QFPay"],
-];
+const providers = {
+    terminal: [
+        ["axepta_bnpp", "pos_iot_worldline", "Axepta BNP Paribas"],
+        ["ingenico", "pos_iot_ingenico", "Ingenico"],
+        ["six_iot", "pos_iot_six", "SIX"],
+        ["adyen", "pos_adyen", "Adyen"],
+        ["mercado_pago", "pos_mercado_pago", "Mercado Pago"],
+        ["razorpay", "pos_razorpay", "Razorpay"],
+        ["stripe", "pos_stripe", "Stripe"],
+        ["viva_com", "pos_viva_com", "Viva.com"],
+        ["worldline", "pos_iot_worldline", "Worldline"],
+        ["tyro", "pos_tyro", "Tyro"],
+        ["pine_labs", "pos_pine_labs", "Pine Labs"],
+        ["qfpay", "pos_qfpay", "QFPay"],
+    ],
+    external_qr: [["payconiq", "pos_payconiq", "Payconiq"]],
+};
```

Comment:
By specifying it here, it will be even easier to use a single variable for both types.